src/osd/PG.cc

```diff
@@ -4011,6 +4027,16 @@ void PG::chunky_scrub(ThreadPool::TPHandle &handle)
 	  scrubber.reserved_peers.clear();
 	}
 
+	{
+	  ObjectStore::Transaction t;
+	  if (scrubber.store) {
+	    scrubber.store->cleanup(&t);
+	  }
```

Comment:
@athanatos thank you. i see. thought the point was all about keeping the shared_ptr around before the remove transaction is applied. will use your version of cleanup.

and i think https://github.com/ceph/ceph/blob/master/src/common/map_cacher.hpp#L161 does not help with anything. if we want to postpone the deleting of keys until the transaction is completed. we should not assign `empty` to `ip`. if, instead, we want to remove the keys from the `MapCacher` right after the `remove_keys()` call, we don't need `vptrs` at all. am i right?